pkg/model/multiversion/delta_test.go

```diff
@@ -0,0 +1,243 @@
+				{
+					fieldName:  "ImageScanningConfiguration",
+					changeType: multiversion.FieldChangeTypeNone,
+				},
+				{
+					fieldName:  "Name",
+					changeType: multiversion.FieldChangeTypeNone,
+				},
+				{
+					fieldName:  "Tags",
+					changeType: multiversion.FieldChangeTypeNone,
+				},
+			},
+		},
+		{
+			name: "v1alpha4-v1alpha3: removed fields.",
+			args: args{
+				spokeVersion: "v1alpha4",
+				hubVersion:   "v1alpha3",
+			},
+			wantErr: false,
+			want: []expectDelta{
+				{
+					fieldName:  "Tags",
+					changeType: multiversion.FieldChangeTypeNone,
+				},
+				{
+					fieldName:  "Name",
+					changeType: multiversion.FieldChangeTypeNone,
+				},
+				{
+					fieldName:  "ImageScanningConfiguration",
+					changeType: multiversion.FieldChangeTypeNone,
+				},
+				{
+					fieldName:  "ITM",
+					changeType: multiversion.FieldChangeTypeNone,
+				},
+				{
+					fieldName:  "AnotherNameField",
+					changeType: multiversion.FieldChangeTypeAdded,
+				},
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			hubGeneratorFile := fmt.Sprintf("generator-%s.yaml", tt.args.hubVersion)
+			hubInferrer := testutil.NewModelForServiceWithOptions(t, "ecr", &testutil.TestingModelOptions{
+				GeneratorConfigFile: hubGeneratorFile,
+				APIVersion:          tt.args.hubVersion,
+			})
+
+			spokeGeneratorFile := fmt.Sprintf("generator-%s.yaml", tt.args.spokeVersion)
+			spokeInferrer := testutil.NewModelForServiceWithOptions(t, "ecr", &testutil.TestingModelOptions{
+				GeneratorConfigFile: spokeGeneratorFile,
+				APIVersion:          tt.args.spokeVersion,
+			})
+
+			hubCRDs, err := spokeInferrer.GetCRDs()
+			require.Nil(err)
+			spokeCRDs, err := hubInferrer.GetCRDs()
+			require.Nil(err)
+			require.Len(hubCRDs, 1)
+			require.Len(spokeCRDs, 1)
+
+			hubCRD := hubCRDs[0]
+			spokeCRD := spokeCRDs[0]
+
+			renames, err := spokeCRD.GetAllRenames()
+			reverseRenames := reverseMap(t, renames)
+
+			require.Nil(err)
+			deltas, err := multiversion.ComputeFieldsDiff(
+				spokeCRD.SpecFields, hubCRD.SpecFields, reverseRenames,
+			)
+
+			if (err != nil) != tt.wantErr {
+				assert.Fail(fmt.Sprintf("multiversion.ComputeFieldsDiff() error = %v, wantErr %v", err, tt.wantErr))
+			}
+			assert.Equal(len(deltas), len(tt.want))
+
+			for i, delta := range deltas {
+				switch delta.ChangeType {
+				case multiversion.FieldChangeTypeRemoved:
+					assert.Equal(tt.want[i].fieldName, delta.Spoke.Names.Camel)
+				case multiversion.FieldChangeTypeAdded:
+					assert.Equal(tt.want[i].fieldName, delta.Hub.Names.Camel)
+				case multiversion.FieldChangeTypeRenamed:
+					assert.Equal(tt.want[i].fieldName, delta.Spoke.Names.Camel)
+				}
+				assert.Equal(tt.want[i].changeType, delta.ChangeType)
+			}
+		})
+	}
+}
+
+func reverseMap(t *testing.T, m map[string]string) map[string]string {
+	reverseMap := make(map[string]string, len(m))
+	for k, v := range m {
+		reverseMap[v] = k
+	}
+	return reverseMap
+}
```

Comment:
bingo. :+1:

Reply:
Removing this utility function and adding `ComputeRenamesDelta` to properly detect the renames from a version to another.